### PR TITLE
More consistent support for students' nicknames

### LINKED ISSUE #3078: Student nickname in Muikku
Nina Kurki: 

> Opiskelijoiden kutsumanimen näkyvyydessä tuntuu olevan vähän epäloogisuutta. Kutsumanimen pitäisi siis toimia siten,että:

>Opiskelijalle on tallennettu kutsumanimi:
Opettajat, ohjaajat ja ylläpitäjät: kaikki nimet "kutsumanimi" sukunimi,
esim. Nina Riitta Johanna "Nina" Kurki

>Opiskelijat (mukaan lukien hän, jonka nimestä kyse): kutsumanimi sukunimi,
esim. Nina Kurki

>Opiskelijalle ei ole tallennettu kutsumanimeä:
Kaikki näkevät nimen muodossa kaikki nimet sukunimi
esim. Nina Riitta Johanna Kurki

>Näkyvyyden pitäisi olla yhteneväinen kaikissa paikoissa:
profiilissa
kirjautumispalkissa
keskustelualueilla
viestimessä
ohjaamossa
arvioinnissa
ym.

>Tällä hetkellä tilanne on sellainen, että:
Opiskelijalle itselleen koko nimi näkyy profiilissa ja kirjautumispalkissa, vaikka kutsumanimi on tallennettu
Kutsumanimi ei näy kummassakaan
Kirjoitettuaan viestin kurssin keskustelualueelle opiskelija näkee, missä muodossa nimi näkyy muille (kutsumanimi+sukunimi).

### PR DESCRIPTION
**Pull request by pkukkon**
- resolves #3078
- nicknames in new evaluation require Pyramus feature 488

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js [hunk 1]
```diff
@@ -418,7 +433,10 @@
             });
           }, this));
           
-          renderDustTemplate('communicator/communicator_items.dust', { items: data, sent: this._folderId == 'sent' }, $.proxy(function (text) {
+          renderDustTemplate('communicator/communicator_items.dust', {
+            items: data, sent: this._folderId == 'sent',
+            isStudent: $('.communicator').attr('data-student') == 'true' ? 1 : ''
```
Comment (ilmoeuro) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js]: why 1 and '' ?

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js [hunk 2]
```diff
@@ -1955,10 +1987,13 @@
           this.setOlderThreadId(thread.olderThreadId);
           this.setNewerThreadId(thread.newerThreadId);
           
-          renderDustTemplate('communicator/communicator_items_open.dust', {messages: messages, labels : labels}, $.proxy(function(text) {
+          renderDustTemplate('communicator/communicator_items_open.dust', {
+            messages: messages,
+            labels: labels,
+            isStudent: $('.communicator').attr('data-student') == 'true' ? 1 : ''
```
Comment (ilmoeuro) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js]: why 1 and '' ?

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js (hunk 1 above)
Comment (ilmoeuro) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js]: Why 1 and '' ?

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js (hunk 2 above)
Comment (ilmoeuro) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js]: Why 1 and '' ?

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js (hunk 1 above)
Comment (pkukkon) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js]: Dust (https://github.com/linkedin/dustjs/wiki/Dust-Tutorial#Logic_in_Templates)

"Take special care if you are trying to pass a boolean parameter. param=true and param=false do not pass true/false as you might expect. They are treated as references to variables named "true" and "false". Unlike JavaScript, they are not reserved names. Note that they are not reserved in JSON either so you can have a property named true or false. So you might think to pass 0 and 1 to your boolean-like parameter. That won't work either. dust's boolean testing ( {?xxx} is more of an existence test than a boolean test. Therefore, with param=1 and param=0 both value exists and so are considered true. Your best bet is to pass 1 and "", e.g. param=1 or param="". You could also leave off param="" if you are sure the name is not elsewhere in your JSON data and accessible."

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js (hunk 2 above)
Comment (pkukkon) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js]: See above

**Pull request merged**